src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs

```diff
@@ -452,6 +452,10 @@ internal async Task ExecuteTarget(ITaskBuilder taskBuilder, BuildRequestEntry re
                         break;
                     }
 
+                    // Don't log the last target finished event until we can process the target outputs as we want to attach them to the
+                    // last target batch. The following statement logs the event for the bucket processed in the previous iteration.
+                    targetLoggingContext?.LogTargetBatchFinished(projectFullPath, targetSuccess, null);
```

Comment:
But is it ok, that we log "Finish" event, before "Start" ?
Can we place it under LogTargetBatchStarted?

Reply:
Actually, no, I would prefer to keep these two calls together so we never reset `targetLoggingContext` to null and we finish the previous bucket just before we start the new one. But I'll add an extra `if` to make the logic clearer.